adapters/Mintegral/MintegralAdapter/GADMediationAdapterMintegral.m

```diff
@@ -4,44 +4,97 @@
-  // TODO: Start signal generation and call completionHandler when the signal generation finishes or
-  // fails.
-  completionHandler("", nil);
+#pragma mark GADRTBAdapter
+
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
+    return [GADMAdapterMintegralExtras class];
 }
 
-+ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
-             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Mintegral SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
++ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration
+             completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
+    NSString *appKey  = @"";
+    NSString *appId = @"";
+    NSError *error = nil;
+    if ([configuration.credentials isKindOfClass:NSArray.class] &&
+        configuration.credentials.count > 0) {
+        appId = configuration.credentials[0].settings[GADMAdapterMintegralAppID];
+        appKey = configuration.credentials[0].settings[GADMAdapterMintegralAppKey];
+        if ([GADMAdapterMintegralUtils isEmpty:appId]) {
+            error = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters,@"AppId is nil");
+            if (completionHandler) {
+                completionHandler(error);
+                return;
+            }
+        }
+        
+        if ([GADMAdapterMintegralUtils isEmpty:appKey]) {
+            error = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters,@"AppKey is nil");
+            if (completionHandler) {
+                completionHandler(error);
+                return;
+            }
+        }
+        
+    }else{
+        error = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters,@"Invalid Server Parameters");
+        if (completionHandler) {
+            completionHandler(error);
+            return;
+        }
+    }
+    
+    // Init MTGSDK
+    [GADMediationAdapterMintegral setAdmobChannel];
+    [[MTGSDK sharedInstance] setAppID:appId ApiKey:appKey];
+    if (completionHandler) {
+        completionHandler(nil);
+    }
+    
 }
 
 + (GADVersionNumber)adSDKVersion {
-  // TODO: Populate `versionString` with the ad-network's SDK version in NSString format.
-  NSString *versionString = @"";
+  NSString *versionString = [MTGSDK sdkVersion];
   GADVersionNumber version = {0};
   NSArray<NSString *> *components = [versionString componentsSeparatedByString:@"."];
-  if (components.count == 4) {
+  if (components.count == 3) {
     version.majorVersion = components[0].integerValue;
     version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue * 100 + components[3].integerValue;
+    version.patchVersion = components[2].integerValue;
```

Comment:
Isn't that Mintegral SDK has 4 digitis?

See https://cocoapods.org/pods/MintegralAdSDK.

Reply:
The [MTGSDK sdkVersion] method can only get the 3-digit version number, do I need to add it to 4 digits?

Reply:
Just to clarify, in CocaoPods there are 4 digits: `7.1.9.0`, also see https://github.com/Mintegral-official/MintegralAdSDK-iOS-new/blob/master/MintegralAdSDK.podspec.

So we should have 4 digits here.

Reply:
Gentle bump on this @Mintegral-Lee.

Reply:
Seems like it's updated to `7.2.1`: https://cocoapods.org/pods/MintegralAdSDK, will the future versions be 3 or 4 digits?

Reply:
Please ignore previous comments,the previous comment was pendding causing the comment to be delayed.

There will only be 3 digits in the future.